lib/src/git.rs

```diff
@@ -500,13 +500,30 @@ fn default_remote_ref_state_for(ref_name: &RefName, git_settings: &GitSettings)
 /// On `import_refs()`, this is similar to collecting commits referenced by
 /// `view.git_refs()`. Main difference is that local branches can be moved by
 /// tracking remotes, and such mutation isn't applied to `view.git_refs()` yet.
-fn pinned_commit_ids(view: &View) -> impl Iterator<Item = &CommitId> {
+fn pinned_commit_ids(view: &View) -> Vec<CommitId> {
     itertools::chain!(
         view.local_branches().map(|(_, target)| target),
         view.tags().values(),
         iter::once(view.git_head()),
     )
     .flat_map(|target| target.added_ids())
+    .cloned()
+    .collect()
+}
+
+/// Commits referenced by untracked remote branches including hidden ones.
```

Comment:
What does it mean for a remote branch to be hidden? Is that equivalent to "deleting" the branch on the remote? Would a deleted branch normally be visible from `view.all_remote_branches()`, or are they present because we haven't finished updating the repo yet?

Reply:
I think this refers to commits which are hidden; I don't think there's a concept of hidden remote branches.

Reply:
Yeah, I used "hidden" as equivalent to "abandoned". Strictly speaking, "hidden" is more correct because "rewritten" commits are also hidden. Updated the other comments to use "hidden".